clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```diff
@@ -1486,41 +1486,40 @@ default DescribeFeaturesResult describeFeatures() {
     UpdateFeaturesResult updateFeatures(Map<String, FeatureUpdate> featureUpdates, UpdateFeaturesOptions options);
 
     /**
-     * Permanently remove a broker and reassign any partitions on the broker.
+     * Unregister a broker.
      * <p>
-     * This operation is supported only on self-managed Kafka clusters (i.e. brokers which do not rely on Zookeeper).
+     * This is a convenience method for {@link #unregisterBroker(int, UnregisterBrokerOptions)}
      *
      * @param brokerId  the broker id to unregister.
      *
-     * <p>This is a convenience method for {@link #decommissionBroker(int, DecommissionBrokerOptions)}
-     *
-     * @return the {@link DecommissionBrokerResult} containing the result
+     * @return the {@link UnregisterBrokerResult} containing the result
      */
-    default DecommissionBrokerResult decommissionBroker(int brokerId) {
-        return decommissionBroker(brokerId, new DecommissionBrokerOptions());
+    default UnregisterBrokerResult unregisterBroker(int brokerId) {
+        return unregisterBroker(brokerId, new UnregisterBrokerOptions());
     }
 
     /**
-     * Permanently remove a broker and reassign any partitions on the broker.
+     * Unregister a broker.
      * <p>
-     * This operation is supported only on self-managed Kafka clusters (i.e. brokers which do not rely on Zookeeper).
+     * This operation does not have any effect on partition assignments. It is supported
+     * only on self-managed Kafka clusters (i.e. brokers which do not rely on Zookeeper).
      *
      * The following exceptions can be anticipated when calling {@code get()} on the future from the
      * returned {@link DescribeFeaturesResult}:
      * <ul>
      *   <li>{@link org.apache.kafka.common.errors.TimeoutException}
      *   If the request timed out before the describe operation could finish.</li>
      *   <li>{@link org.apache.kafka.common.errors.UnsupportedVersionException}
-     *   If the software is too old to support decommissioning.
+     *   If the software is too old to support the unregistration API.
      * </ul>
      * <p>
      *
      * @param brokerId  the broker id to unregister.
      * @param options   the options to use.
      *
-     * @return the {@link DecommissionBrokerResult} containing the result
+     * @return the {@link UnregisterBrokerResult} containing the result
      */
-    DecommissionBrokerResult decommissionBroker(int brokerId, DecommissionBrokerOptions options);
+    UnregisterBrokerResult unregisterBroker(int brokerId, UnregisterBrokerOptions options);
```

Comment:
@cmccabe We should mark all KIP-500 methods with `Unstable` and mention it in the documentation too.